### PR TITLE
Code to Convert any 40x11 Translucent Gray-Scaled Bitmap to Hex Arrays

### LINKED ISSUE #17: Add feature to display bitmap
Please implement the possibility to send a Bitmap to the LED badge directly. Since we have a full control over the badge device, we can create a 40x11px bitmap and write some code to convert it into byte arrays we can send to the LED badge.

screenshot from 2018-12-17 08-00-16

09_final

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #17 
Changes: 
- a singleton to convert drawable to hex arrays

Screenshots for the change:
Screenshot_20190325-211831_Badge Magic
Screenshot_20190326-105313_Badge Magic

### CODE REVIEW

**Comment by adityastic** [comment]
> A lot of magic numbers are used. Extract into constants and variables

Sure Sir

**Comment by adityastic** [comment]
@iamareebjamal Done, Have a look :)

File: app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        val blackValue = -16777216
+
+        val image = Array(height) { IntArray(width) }
+        for (i in 0..(height - 1)) {
+            for (j in 0..(width - 1)) {
+                image[i][j] = if (bm.getPixel(j, i) == blackValue) 1 else 0
+            }
+        }
+        var finalSum = 0
+        for (j in 0..(width - 1)) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt]: Use until

File: app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                for (i in 0..(height - 1)) {
+                    image[i][j] = -1
+                }
+            } else {
+                finalSum += j
+                break
+            }
+        }
+
+        for (j in (width - 1) downTo -1 + 1) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt]: `(width - 1) downTo -1 + 1`?

File: app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            for (j in 0 until lOff) {
+                list[i].add(0)
+            }
+        }
+        val allHexs = ArrayList<String>()
+        for (i in 0 until list[0].size / 8) {
+            val lineHex = StringBuilder()
+            for (k in 0..(height - 1)) {
+
+                val `as` = StringBuilder()
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt]: Why name a variable as keyword?

File: app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt [hunk 4]
```diff
@@ -0,0 +1,98 @@
+package com.nilhcem.blenamebadge.util
+
+import android.content.Context
+import android.graphics.drawable.BitmapDrawable
+import java.math.BigInteger
+import java.util.ArrayList
+
+object Converters {
+    fun convertImageToLEDHex(context: Context, dId: Int): ArrayList<String> {
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt]: Return List<String>

Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt]: Use @DrawableRes annotation

File: app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt (hunk 2 above)
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt]: -1 + 1. LOLOL, Im so dumb

**Comment by adityastic** [comment]
@iamareebjamal Fixed, Have a look.

File: app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    fun convertImageToLEDHex(context: Context, @DrawableRes dId: Int): List<String> {
+        val myIcon = context.resources.getDrawable(dId)
+        val bm = (myIcon as BitmapDrawable).bitmap
+
+        val height = bm.height
+        val width = bm.width
+        val blackValue = -16777216
+
+        val image = Array(height) { IntArray(width) }
+        for (i in 0 until (height - 1)) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt]: Using until should not need `(height - 1)`? That's why I recommended that

Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt]: Gotcha

**Comment by adityastic** [comment]
@iamareebjamal Done, fixed all the until's

**Pull request merged**